src/common/http_defs.cc

```diff
@@ -131,7 +131,21 @@ namespace Pistache::Http
         case Type::RFC1123GMT: {
             // Requires GMT so we must use std::gmtime to convert to GMT
             time_t t = std::chrono::system_clock::to_time_t(date_);
-            os << std::put_time(std::gmtime(&t), "%a, %d %b %Y %T %Z");
+            
+            // July/2024. For a std::tm* that comes from std::gmtime,
+            // std::put_time %Z should always outout the std::tm locale namely
+            // "GMT" (or "UTC"). However, in NetBSD 10.0, std::put_time appears
+            // to output the locale of the machine it's running on ("PST") not
+            // the locale of the std::tm*. We workaround as per the following
+            // #defines. Since we know the locale of std::gmtime is GMT, the
+            // workaround is correct in general.
+            os << std::put_time(std::gmtime(&t), "%a, %d %b %Y %T "
+#ifdef __NetBSD__
+                                "GMT"
+#else                                
+                                "%Z"
+#endif
```

Comment:
Mmmm. Interesting analysis.

I was relying heavily on the example here:
https://en.cppreference.com/w/cpp/chrono/c/gmtime

But you're right, the example is relying on std::tm's tm_isdst field, *not* on the tm_zone field.

https://en.cppreference.com/w/cpp/chrono/c/tm states:
BSD, GNU and musl C library support two additional members [including tm_zone], which are standardized in POSIX.1 2024.

> My guess is that NetBSD never fully implemented the GNU extensions, and doesn't yet implement POSIX 2024, so does the perfectly legal thing of printing the time zone abbreviation associated with the current locale, instead of the one associated with the tm object.
[DG] That sounds right

> I've opened https://github.com/pistacheio/pistache/pull/1222 to fix this, so that it can be merged independently of this port. Once merged, be sure to git rebase on that!
[DG] Happy to, as soon as there's a different fix.

Reply:
Meanwhile, I am updating my comment in http_defs.cc.